lib/compat/wordpress-6.5/class-gutenberg-rest-global-styles-revisions-controller-6-5.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Global_Styles_Revisions_Controller class, inspired by WP_REST_Revisions_Controller.
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.3.0
+ */
+
+/**
+ * Core class used to access global styles revisions via the REST API.
+ *
+ * @since 6.3.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Global_Styles_Revisions_Controller_6_5 extends Gutenberg_REST_Global_Styles_Revisions_Controller_6_4 {
+	/**
+	 * Registers the controller's routes.
+	 *
+	 * @since 6.3.0
+	 * @since 6.5.0 Adds route to fetch individual global styles revisions.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->parent_base . '/(?P<parent>[\d]+)/' . $this->rest_base . '/(?P<id>[\d]+)',
+			array(
+				'args'   => array(
+					'parent' => array(
+						'description' => __( 'The ID for the parent of the global styles revision.' ),
+						'type'        => 'integer',
+					),
+					'id'     => array(
+						'description' => __( 'Unique identifier for the global styles revision.' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+		parent::register_routes();
+	}
+
+	/**
+	 * Retrieves one global styles revision from the collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$parent = $this->get_parent( $request['parent'] );
+		if ( is_wp_error( $parent ) ) {
+			return $parent;
+		}
+
+		$revision = $this->get_revision( $request['id'] );
+		if ( is_wp_error( $revision ) ) {
+			return $revision;
+		}
+
+		$response = $this->prepare_item_for_response( $revision, $request );
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Get the global styles revision, if the ID is valid.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param int $id Supplied ID.
+	 * @return WP_Post|WP_Error Revision post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_revision( $id ) {
+		$error = new WP_Error(
+			'rest_post_invalid_id',
+			__( 'Invalid revision ID.' ),
+			array( 'status' => 404 )
+		);
+
+		if ( (int) $id <= 0 ) {
+			return $error;
+		}
+
+		$revision = get_post( (int) $id );
+		if ( empty( $revision ) || empty( $revision->ID ) || 'revision' !== $revision->post_type ) {
+			return $error;
+		}
```

Comment:
Do we need to check that the `$revision->post_parent` matches the provided parent id? Currently it looks like the endpoint might only be checking that the parent is valid and can be retrieved, not that the revision id matches that parent 🤔

Reply:
This is also taken from `WP_REST_Revisions_Controller`. 

The parent post type check (`'wp_global_styles'`) should be enough for our purposes to start with.

You bring up a great point though. 

Currently I can, for any post type, return the revision of different post just by knowing the post ID.

So for the following:

`await wp.apiFetch( { path: '/wp/v2/posts/19/revisions/40' } )`

`19` is the id of a different parent. So long as `19` is valid and `40` is a revision post type, I'll get the revision object of `40`.

<img width="782" alt="Screenshot 2023-11-08 at 5 32 20 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/bc2255a3-3d33-442b-a6e5-cd19252cb6ee">

As with [the above comment](https://github.com/WordPress/gutenberg/pull/55827/files#r1385973042), it could be a good question to raise in Core Slack or in a trac ticket. 

There might be a reason why all post type revisions work this way. 🤔 

I'll make a note to ask. Thank you!

Reply:
Question asked here: https://wordpress.slack.com/archives/C02RQC26G/p1699504213232439